src/Plugin.cpp

```diff
@@ -0,0 +1,296 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "Plugin.hpp"
+
+#include "../../../src/cs-core/GuiManager.hpp"
+#include "../../../src/cs-core/Settings.hpp"
+#include "../../../src/cs-utils/logger.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+#include "logger.hpp"
+
+#include <CivetServer.h>
+#include <GL/glew.h>
+#include <VistaKernel/DisplayManager/VistaDisplayManager.h>
+#include <VistaKernel/DisplayManager/VistaWindow.h>
+#include <VistaKernel/VistaFrameLoop.h>
+#include <VistaKernel/VistaSystem.h>
+#include <curlpp/cURLpp.hpp>
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb_image_write.h>
+#include <utility>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+EXPORT_FN cs::core::PluginBase* create() {
+  return new csp::webapi::Plugin;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+EXPORT_FN void destroy(cs::core::PluginBase* pluginBase) {
+  delete pluginBase; // NOLINT(cppcoreguidelines-owning-memory)
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Converts a void* to a std::vector<char> (which is given through a void* as well). So this is
+// pretty unsafe, but I think it's the only way to make stb_image write to a std::vector<char>. If
+// anybody has a better idea...
+void pngWriteToVector(void* context, void* data, int len) {
+  // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
+  auto vector = reinterpret_cast<std::vector<char>*>(context);
+  // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
+  auto charData = reinterpret_cast<char*>(data);
```

Comment:
I think you can use `static_cast` here since `void*` and `char*` are convertible to each other.

```suggestion
  auto* charData = static_cast<char*>(data);
```